### PR TITLE
feat(buttons): add X social button

### LINKED ISSUE #2182: Drop 'Twitter' and introduce 'X'?
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

According to recent news, 'Twitter' (https://twitter.com/) has been rebranded to 'X'.

screenshot-publish twitter com-2023 07 31-09_33_31

Actually it looks like a bit messy and it's still not clear if it is temporary or not. I don't even know if we still legally have the right to use the name 'Twitter' and/or the logo with the bird inside Boosted's documentation:

twitter

### Motivation and context

What are your thoughts on this?
Is there a need to do something here?
Should we keep the social button with the bird only? Replace it by a 'X' button? Or maybe keep both alternatives?

@B3nz01d @julien-deramond @CyriaqueBillard @Franco-Riccitelli

### PR DESCRIPTION
**Pull request by julien-deramond**
### Related issues

Closes #2182

### Description

This PR adds a new social button for the X social media platform.
We chose not to get rid of the Twitter one for now since some websites prefer to still use it.

### Types of change

- New feature (non-breaking change which adds functionality)

### Live previews

- Social buttons: https://deploy-preview-2644--boosted.netlify.app/docs/5.3/components/buttons/#social-buttons
- Footer with all social icons in dark and light modes: https://deploy-preview-2644--boosted.netlify.app/docs/5.3/dark-mode/#footer
- Migration guide: https://deploy-preview-2644--boosted.netlify.app/docs/5.3/migration/#buttons

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -1078,6 +1078,10 @@ $btn-social-networks: (
   "tiktok": (
     "color": #ff2c55,
     "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32'><path d='M7.024 30.054C4.584 28.212 3 25.235 3 21.876c0-5.59 4.39-10.123 9.805-10.123.45 0 .899.031 1.345.094v5.6a4.363 4.363 0 0 0-1.361-.218c-2.477 0-4.485 2.074-4.485 4.631 0 1.809 1.003 3.374 2.467 4.137l.31.146a4.348 4.348 0 0 0 1.708.348c2.471 0 4.476-2.065 4.484-4.615V0h5.335v.704c.02.211.046.42.082.63l.08.404a7.668 7.668 0 0 0 3.306 4.769A7.22 7.22 0 0 0 30 7.665V8.83l-.199-.047-.182-.047.381.094v4.312a12.4 12.4 0 0 1-7.392-2.443v11.177c0 5.591-4.39 10.124-9.804 10.124-2.02 0-3.898-.63-5.458-1.712l-.322-.234Z'/></svg>"
+  ),
+  "x": (
+    "color": #1da1f2,
+    "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 19'><path d='M0.0486667 0L7.77044 10.3247L0 18.7191H1.74889L8.552 11.3696L14.0487 18.7191H20L11.8438 7.81355L19.0764 0H17.3276L11.0624 6.76867L6.00022 0H0.0486667ZM2.62067 1.28822H5.35467L17.4278 17.4307H14.6938L2.62067 1.28822Z'/></svg>"
```
Comment (louismaximepiton) [inline scss/_variables.scss:1084]: Just ran svgo on this: 
```suggestion
    "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 19'><path d='M.049 0 7.77 10.325 0 18.719h1.749l6.803-7.35 5.497 7.35H20L11.844 7.814 19.076 0h-1.748l-6.266 6.769L6 0H.05ZM2.62 1.288h2.734l12.073 16.143h-2.734L2.62 1.288Z'/></svg>"
```

And just used the logo from [their documentation](https://about.x.com/en/who-we-are/brand-toolkit): 
```suggestion
    "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1200 1227'><path d='M714.163 519.284 1160.89 0h-105.86L667.137 450.887 357.328 0H0l468.492 681.821L0 1226.37h105.866l409.625-476.152 327.181 476.152H1200L714.137 519.284h.026ZM569.165 687.828l-47.468-67.894-377.686-540.24h162.604l304.797 435.991 47.468 67.894 396.2 566.721H892.476L569.165 687.854v-.026Z'/></svg>"
```

File: scss/_variables.scss [hunk 2]
```diff
@@ -1078,6 +1078,10 @@ $btn-social-networks: (
   "tiktok": (
     "color": #ff2c55,
     "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32'><path d='M7.024 30.054C4.584 28.212 3 25.235 3 21.876c0-5.59 4.39-10.123 9.805-10.123.45 0 .899.031 1.345.094v5.6a4.363 4.363 0 0 0-1.361-.218c-2.477 0-4.485 2.074-4.485 4.631 0 1.809 1.003 3.374 2.467 4.137l.31.146a4.348 4.348 0 0 0 1.708.348c2.471 0 4.476-2.065 4.484-4.615V0h5.335v.704c.02.211.046.42.082.63l.08.404a7.668 7.668 0 0 0 3.306 4.769A7.22 7.22 0 0 0 30 7.665V8.83l-.199-.047-.182-.047.381.094v4.312a12.4 12.4 0 0 1-7.392-2.443v11.177c0 5.591-4.39 10.124-9.804 10.124-2.02 0-3.898-.63-5.458-1.712l-.322-.234Z'/></svg>"
+  ),
+  "x": (
+    "color": #1da1f2,
```
Comment (louismaximepiton) [inline scss/_variables.scss:1083]: It's the right color regarding the UI Kit, but are we sure that this color is still good from the X pov ?

**Review by louismaximepiton — COMMENTED** [review]
The look is already fine, just quick questions

File: scss/_variables.scss (hunk 1 above)
Comment (julien-deramond) [reply scss/_variables.scss:1084]: We have to use the version of the logo defined in our design kit.

File: scss/_variables.scss (hunk 2 above)
Comment (julien-deramond) [reply scss/_variables.scss:1083]: I don't think so, but without any further information, I've used this one. Should be black or grayish IMO.

File: scss/_variables.scss [hunk 3]
```diff
@@ -1078,6 +1078,10 @@ $btn-social-networks: (
...
+    "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 19'><path d='M.049 0 7.77 10.325 0 18.719h1.749l6.803-7.35 5.497 7.35H20L11.844 7.814 19.076 0h-1.748l-6.266 6.769L6 0H.05ZM2.62 1.288h2.734l12.073 16.143h-2.734L2.62 1.288Z'/></svg>"
```
Comment (julien-deramond) [inline scss/_variables.scss:1084]:
```suggestion
    "icon": "<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 19'><path d='M0.0486667 0L7.77044 10.3247L0 18.7191H1.74889L8.552 11.3696L14.0487 18.7191H20L11.8438 7.81355L19.0764 0H17.3276L11.0624 6.76867L6.00022 0H0.0486667ZM2.62067 1.28822H5.35467L17.4278 17.4307H14.6938L2.62067 1.28822Z'/></svg>"
```

**Comment by louismaximepiton** [comment]
How do the designers make the logo for this kind of component? It seems lighter than the logo from [their documentation](https://about.x.com/en/who-we-are/brand-toolkit) from a size point of view but isn't it altered then? The question is up to you.

**Comment by CyriaqueBillard** [comment]
@julien-deramond @louismaximepiton As for the hover state of the X social button, we should be using the same blue as the blue used for the hover state of the X button on browser. If you are able to pick it up, I will change it in the UI kit, so we'll be aligned.

**Comment by Franco-Riccitelli** [comment]
I have updated and published the X logo used in the social button - in the Figma Web UI kit. The previous version was very thin and looked different to the standard version that appears on X.

**Comment by Franco-Riccitelli** [comment]
The previews look good to me for the Social buttons; however, if it is possible to update the X logo to the version I have added to the Figma Web UI kit, that would be great.
As already mentioned, the Figma Web UI kit can be updated with the Blue hover state colour used in the preview `#1da1f2`, 
 so no need to change this on the Dev side.

**Comment by julien-deramond** [comment]
Thanks for your feedback.
* @Franco-Riccitelli, I have updated the logo based on the new one provided in the kit:

| Before | Now |
|---|---|
| Screenshot 2024-07-25 at 07 02 28 | Screenshot 2024-07-25 at 07 02 33 |

* @CyriaqueBillard @Franco-Riccitelli, based on your 2 comments, I haven't changed the blue hover color and I let you update it whenever it's possible with the value used here in the code. As spotted by Franco, this is `#1da1f2`.
* @louismaximepiton, the logo SVG has been transformed via svgo

**Comment by Franco-Riccitelli** [comment]
Update looks good to me.

**Review by louismaximepiton — APPROVED** [review]
:shipit:

**Pull request merged**